packages/contracts/src/contracts/current/test/Whitelist/Whitelist.sol

```diff
@@ -0,0 +1,80 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.21;
+pragma experimental ABIEncoderV2;
+
+import "../../protocol/Exchange/Exchange.sol";
+import "../../protocol/Exchange/LibOrder.sol";
+import "../../utils/Ownable/Ownable.sol";
+
+contract Whitelist is Ownable {
+
+    mapping (address => bool) public isWhitelisted;
+    Exchange EXCHANGE;
+
+    bytes txOriginSignature = new bytes(1);
+    bytes4 fillOrderFunctionSelector;
+
+    function Whitelist(address _exchange)
+        public
+    {
+        EXCHANGE = Exchange(_exchange);
+        txOriginSignature[0] = 0x04;
+        fillOrderFunctionSelector = EXCHANGE.fillOrder.selector;
+    }
+
+    function updateWhitelistStatus(address target, bool isApproved)
+        external
+        onlyOwner
+    {
+        isWhitelisted[target] = isApproved;
+    }
+
+    function fillOrderIfWhitelisted(
```

Comment:
Could be good to add function descriptions. We've been including them in the newer contracts.
An example for this function:
```
/// @dev Fills the input order using msg.sender as the order taker.
         The sender must be both whitelisted and the originating address of this transaction.
         This function throws if the sender does not meet these requirements or the order cannot be filled.
/// @param order Order struct containing order specifications.
/// @param takerAssetFillAmount Desired amount of takerToken to sell.
/// @param salt Arbitrary number to ensure uniqueness of transaction.
/// @param orderSignature Proof that order has been created by maker.
```